src/pages/Login/login.js

```diff
@@ -0,0 +1,61 @@
+import Loader from '../../Components/Loader/Loader';
+import ImageComponent from './ImageComponent/ImageComponent'
+import Logo from './Logo/Logo';
+import CustomInput from './CustomInput/CustomInput'
+import { useState } from 'react';
+import CustomButton from './CustomButton/CustomButton';
+
+export default function Login() {
+  const [emailAddress, setEmailAddress] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+  const isInvalid = password === '' || emailAddress === '';
+  const handleLogin = async (event) => {
+    event.preventDefault();
+
+    try {
+      
+    } catch (error) {
+      setEmailAddress('');
+      setPassword('');
+      setError(error.message);
+    }
+  };
+  const buttonProps = {
+    disabled: isInvalid,
+    type: 'submit',
+    styleButton: `bg-blue-medium text-white w-full rounded h-8 font-bold ${isInvalid && 'opacity-50'}`,
+  };
+  return (
+    <div className="container flex mx-auto max-w-screen-md items-center h-screen">
+       <Loader fadeOutInterval={5000}/>
+       <ImageComponent src="/images/iphone-with-profile.jpg"  alt="iPhone with Instagram app" styleImg='flex w-3/5' />
+       <div className="flex flex-col w-2/5">
+        <div className="flex flex-col items-center bg-white p-4 border border-gray-primary mb-4 rounded">
+          
+        <Logo src="/images/logo.png" alt="Instagram" style1="flex justify-center w-full"  style2="mt-2 w-6/12 mb-4" />
+        {error && <p className="mb-4 text-xs text-red-primary">{error}</p>}
+        <form onSubmit={handleLogin} method="POST">
+            <CustomInput
+              aria-label="Enter your email address"
+              type="text"
+              placeholder="Email address"
+              styleInput="text-sm text-gray-base w-full mr-3 py-5 px-4 h-2 border border-gray-primary rounded mb-2"
+              onChange={({ target }) => setEmailAddress(target.value)}
+              value={emailAddress}
+            />
+            <CustomInput
+              aria-label="Enter your password"
+              type="password"
+              placeholder="Password"
+              styleInput ="text-sm text-gray-base w-full mr-3 py-5 px-4 h-2 border border-gray-primary rounded mb-2"
+              onChange={({ target }) => setPassword(target.value)}
+              value={password}
+            />
+            <CustomButton {...buttonProps}>Login</CustomButton>
```

Comment:
Is there a reason for passing the word "Login" as a child of the CustomButton component instead of passing it from the props?

Reply:
In the this code, the "Login" text is passed as a child of the CustomButton component rather than being passed as a prop. The reason for this approach could be to allow for more flexibility in the usage of the CustomButton component.

By passing the "Login" text as a child, the component can be used not only for login functionality but also for other cases where a button with different text content is needed. This allows the component to be more reusable and versatile.

If the "Login" text were passed as a prop instead, the component would be limited to only displaying that specific text, and it might require creating a separate component or prop for different button labels.